### ISSUE TITLE
Macro Learn mapping font size is only adapted when showing view

### ISSUE THREAD

**Issue by pdesaulniers** [zynaddsubfx/zyn-fusion-issues#175]
I'm using latest zyn built with `build-linux.rb`. 

Here's how to reproduce the issue:
- Click on Macro Learn. Here, I had Slot 1 already selected. Notice the mapping.
![2018-07-13-15 46 18](https://user-images.githubusercontent.com/23723605/42711014-660a3e4a-86b4-11e8-8cac-d41809710eb9.png)
- Select Slot 0, which has a bigger mapping. The text is spilling over the slot names:
![2018-07-13-15 50 43](https://user-images.githubusercontent.com/23723605/42711176-ecf950da-86b4-11e8-8884-614c16145ecf.png)
- Leave Macro Learn menu, then come back. The font size is OK now:
![2018-07-13-15 51 50](https://user-images.githubusercontent.com/23723605/42711108-c5646ab4-86b4-11e8-85ea-a2a8922ea98b.png)

I guess the font size should be adapted when switching slots too.

**Comment by fundamental** [comment]
Yeah, that makes sense. Essentially the layout phase would change the box to something reasonable when going back to the view, but if the text is updated while in the view no additional layout is triggered resulting in an widget which has a valid bounding box, but drawing routines which exceed the limit for the bounding box.

The short term fix would be adding a check to the text label's drawing routine to verify it doesn't exceed the bounding box and to change text height if need be.

The long term fix is going to be the more pleasant rendering of the path with something like "P0-K0-Add-filter-base fq."
